File: week_01/Python_code/strStr.py
class Solution:
    def strStr(self, source, target):
        if target is None or source is None:
            return -1
        elif len(target) == 0:
            return 0
        elif len(source) == 0 or len(target) > len(source):
            return -1
        for i in xrange(len(source) - len(target) + 1):
            temp = i
            for j in xrange(0, len(target)):
                if source[temp + j] != target[j]:
                    break
                if j == len(target) - 1:
                    return temp
        return -1

if __name__ == '__main__':
    test = Solution()
    source = 'abcdabcdefg'
    target = 'bcde'
    print test.strStr(source, target)

File: week_01/Java_Code/StrStr.java
public class StrStr {
    /**
     * Returns a index to the first occurrence of target in source,
     * or -1  if target is not part of source.
     * @param source string to be scanned.
     * @param target string containing the sequence of characters to match.
     */
    public int strStr(String source, String target) {
        if (source == null || target == null) {
            return -1;
        }
        else if (target.length() == 0) {
            return 0;
        }
        else if (source.length() == 0 || source.length() < target.length()) {
            return -1;
        }
        int i, j;
        for (i = 0; i < source.length() - target.length() + 1; i++){
            int temp = i;
            for (j = 0; j < target.length(); j++) {
                if (source.charAt(temp + j) != target.charAt(j)) {
                    break;
                }
            }
            if (j == target.length()) {
                    return temp;
                }
        }
        return -1;
    }

    public static void main(String[] args) {
        String source = "abcdbcdefg";
        String target = "bcde";
        StrStr test = new StrStr();
        int result = test.strStr(source, target);
        System.out.println(result);
    }
}